boa/src/value/tests.rs

```diff
@@ -64,62 +64,47 @@ fn number_is_true() {
-    assert_eq!(forward(&mut context, "[1,2] == '1,2'"), "true");
-    assert_eq!(forward(&mut context, "new String('foo') == 'foo'"), "true");
-    assert_eq!(forward(&mut context, "null == undefined"), "true");
-    assert_eq!(forward(&mut context, "undefined == null"), "true");
-    assert_eq!(forward(&mut context, "null == false"), "false");
-    assert_eq!(forward(&mut context, "[] == ![]"), "true");
-    assert_eq!(
-        forward(
-            &mut context,
-            "a = { foo: 'bar' }; b = { foo: 'bar'}; a == b"
-        ),
-        "false"
-    );
-    assert_eq!(
-        forward(&mut context, "new String('foo') == new String('foo')"),
-        "false"
-    );
-    assert_eq!(forward(&mut context, "0 == null"), "false");
-
-    assert_eq!(forward(&mut context, "0 == '-0'"), "true");
-    assert_eq!(forward(&mut context, "0 == '+0'"), "true");
-    assert_eq!(forward(&mut context, "'+0' == 0"), "true");
-    assert_eq!(forward(&mut context, "'-0' == 0"), "true");
-
-    assert_eq!(forward(&mut context, "0 == NaN"), "false");
-    assert_eq!(forward(&mut context, "'foo' == NaN"), "false");
-    assert_eq!(forward(&mut context, "NaN == NaN"), "false");
-
-    assert_eq!(
-        forward(
-            &mut context,
-            "Number.POSITIVE_INFINITY === Number.POSITIVE_INFINITY"
-        ),
-        "true"
-    );
-    assert_eq!(
-        forward(
-            &mut context,
-            "Number.NEGAVIVE_INFINITY === Number.NEGAVIVE_INFINITY"
-        ),
-        "true"
+    check_output(
+        None,
+        &[
+            ("undefined == undefined", "true"),
+            ("null == null", "true"),
+            ("true == true", "true"),
+            ("false == false", "true"),
+            ("'foo' == 'foo'", "true"),
+            ("0 == 0", "true"),
+            ("+0 == -0", "true"),
+            ("+0 == 0", "true"),
+            ("-0 == 0", "true"),
+            ("0 == false", "true"),
+            ("'' == false", "true"),
+            ("'' == 0", "true"),
+            ("'17' == 17", "true"),
+            ("[1,2] == '1,2'", "true"),
+            ("new String('foo') == 'foo'", "true"),
+            ("null == undefined", "true"),
+            ("undefined == null", "true"),
+            ("null == false", "false"),
+            ("[] == ![]", "true"),
+            ("a = { foo: 'bar' }; b = { foo: 'bar'}; a == b", "false"),
+            ("new String('foo') == new String('foo')", "false"),
+            ("0 == null", "false"),
+            ("0 == '-0'", "true"),
+            ("0 == '+0'", "true"),
+            ("'+0' == 0", "true"),
+            ("'-0' == 0", "true"),
+            ("0 == NaN", "false"),
+            ("'foo' == NaN", "false"),
+            ("NaN == NaN", "false"),
+            (
+                "Number.POSITIVE_INFINITY === Number.POSITIVE_INFINITY",
+                "true",
+            ),
+            (
+                "Number.NEGAVIVE_INFINITY === Number.NEGAVIVE_INFINITY",
```

Comment:
```suggestion
                "Number.NEGATIVE_INFINITY === Number.NEGATIVE_INFINITY",
```
Realized there's a typo here 😅, this is evaluating `undefined === undefined`.